network/common/src/main/java/org/apache/spark/network/util/JavaUtils.java

```diff
@@ -121,4 +121,59 @@ private static boolean isSymlink(File file) throws IOException {
     }
     return !fileInCanonicalDir.getCanonicalFile().equals(fileInCanonicalDir.getAbsoluteFile());
   }
+
+  /**
+   * Convert a passed time string (e.g. 50s, 100ms, or 250us) to a microsecond count for
+   * internal use. If no suffix is provided a direct conversion is attempted.
+   */
+  private static long timeStringToUs(String str) throws NumberFormatException {
```

Comment:
This method can be very similar to the Scala one (using an `com.google.common.collect.ImmutableMap` to hold the valid suffixes). Maybe that's what you meant by updating the Java code later.